### PR TITLE
[Needs PO review] Add missing permissions checks to background jobs

### PR DESCRIPTION
**Pull request by jcoyne**
Ref #1491

### CODE REVIEW

**Comment by mjgiarlo** [comment]
@jcoyne 	There are some conflicts to resolve on this PR now.

**Comment by ndushay** [comment]
I think it's also worth doing a smoke test in stage -- impersonate a group that isn't enabled, make sure it correctly doesn't allow stuff;  impersonate a group that is enabled, make sure it does allow stuff.

Also, since this may affect argo users that aren't in the right groups, I think we should  do a PO review @andrewjbtw

File: app/jobs/virtual_merge_job.rb [hunk 1]
```diff
@@ -1,20 +1,30 @@
 # frozen_string_literal: true
 
 ##
-# job to merge objects
+# job to merge objects and then close the version
 class VirtualMergeJob < GenericJob
   queue_as :default
 
   def perform(_bulk_action_id, parent_druid:, child_druids:)
+    return unless authorized_to_manage?(parent_druid)
```
[the remark this answers is not in the document]
Comment (andrewjbtw) [reply app/jobs/virtual_merge_job.rb]: Now that I understand better what's happening behind the scenes with virtual objects, I think if the user has permission to manage the parent, then it shouldn't be necessary to check the child objects. So my answer is no.

**Comment by andrewjbtw** [comment]
I believe this is working correctly, from what I've been able to determine on stage. I think we should go ahead and watch carefully to see if users experience a problem running background jobs.

File: app/jobs/close_version_job.rb [hunk 2]
```diff
@@ -10,11 +10,10 @@ class CloseVersionJob < GenericJob
   # @param [Integer] bulk_action_id GlobalID for a BulkAction object
   # @param [Hash] params additional parameters that an Argo job may need
   # @option params [Array] :pids required list of pids
-  # @option params [Array] :groups the groups the user belonged to when the started the job. Required for permissions check (can_manage?)
+  # @option params [Array] :groups the groups the user belonged to when the started the job. Required for because groups are not persisted with the user.
```
Comment (mjgiarlo) [inline app/jobs/close_version_job.rb]: `s/for because/because/`

File: app/jobs/modsulator_job.rb [hunk 3]
```diff
@@ -16,18 +16,21 @@ class ModsulatorJob < ActiveJob::Base
   # @param  [String]  apo_id             DRUID of the DOR APO that governs all of the objects we're trying to upload metadata for.
   # @param  [String]  uploaded_filename  Full path to the temporary uploaded file. Deleted upon completion.
   # @param  [String]  output_directory   Where to store output (log, generated XML etc.).
-  # @param  [String]  user_login         Acting user's username.
+  # @param  [User]    user               Acting user
   # @param  [String]  filetype           One of 'xml, 'spreadsheet', or 'xml_only'. If not 'xml', the input is to be loaded into the datastream.
   # @param  [String]  note               An optional note that the user entered to go with the job.
   # @return [Void]
-  def perform(apo_id, uploaded_filename, output_directory, user_login, filetype = 'spreadsheet', note = '')
+  def perform(apo_id, uploaded_filename, output_directory, user, groups, filetype = 'spreadsheet', note = '')
```
Comment (mjgiarlo) [inline app/jobs/modsulator_job.rb]: Do we also need to add `groups` as a `@param` above?

File: app/jobs/modsulator_job.rb [hunk 4]
```diff
@@ -124,13 +127,13 @@ def generate_original_filename(uploaded_filename)
 
   # Write initial job information to the log file.
   #
-  # @param [File]    log_file  The log file to write to.
-  # @param [String]  username  The login name of the current user.
-  # @param [String]  filename  The name of this job's input file.
-  # @param [String]  note      An optional comment that describes this job.
-  def start_log(log_file, username, filename, note = '')
+  # @param [File] log_file The log file to write to.
+  # @param [user] username The login name of the current user.
```
Comment (mjgiarlo) [inline app/jobs/modsulator_job.rb]: I think this should be `@param [User] user` instead of `@param [user] username`?

**Comment by mjgiarlo** [comment]
Up to @jcoyne 	whether he wants to act on my code review. Nothing big there. Merge when ready.

**Comment by ndushay** [comment]
I'm gonna merge this to get it out, but leave @mjgiarlo and @jcoyne to do a followon PR for stuff Mike mentioned.

**Pull request merged**